### PR TITLE
Update/type demo

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| Infrastructure |
|**_Affected Component_**| Complex Types |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  #1913 |
|**_Has Unit Tests (y/n)_**| n  |
|**_Builds Without Errors (y/n)_**| y |
|**_Unit Tests Pass (y/n)_**| y |
|**_Documentation Included (y/n)_**| n |

---
## Change Description

Adds in a type demo component that has complex commands and parameters!

## Rationale

To support complex commanding we need a component that has such commands and parameters.

## Testing/Review Recommendations

Test with: fprime-community/fprime-gds#122

## Future Work

Make recursive dictionary generation work (complete fix for #1913)

### CODE REVIEW

**Comment by thomas-bc** [comment]
@LeStarch cleanest way to get CppCheck and CppLint scans to run will be to rebase locally on top of `devel` and push the changes to your PR, I think.

**Comment by bocchino** [comment]
This is very cool!

File: Ref/TypeDemo/TypeDemo.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+void TypeDemo ::CHOICES_cmdHandler(const FwOpcodeType opCode, const U32 cmdSeq, Ref::ManyChoices choices) {
+    this->tlmWrite_ChoicesCh(choices);
+    this->log_ACTIVITY_HI_ChoicesEv(choices);
+    this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+}
+
+void TypeDemo ::CHOICES_WITH_FRIENDS_cmdHandler(const FwOpcodeType opCode,
+                                                const U32 cmdSeq,
+                                                U8 repeat,
```
Comment (timcanham) [inline Ref/TypeDemo/TypeDemo.cpp:41]: repeat is never used?

File: Ref/TypeDemo/TypeDemo.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+void TypeDemo ::EXTRA_CHOICES_cmdHandler(const FwOpcodeType opCode, const U32 cmdSeq, Ref::TooManyChoices choices) {
+    this->tlmWrite_ExtraChoicesCh(choices);
+    this->log_ACTIVITY_HI_ExtraChoicesEv(choices);
+    this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+}
+
+void TypeDemo ::EXTRA_CHOICES_WITH_FRIENDS_cmdHandler(const FwOpcodeType opCode,
+                                                      const U32 cmdSeq,
+                                                      U8 repeat,
```
Comment (timcanham) [inline Ref/TypeDemo/TypeDemo.cpp:59]: repeat is never used?

File: Ref/TypeDemo/TypeDemo.cpp [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+void TypeDemo ::CHOICE_PAIR_cmdHandler(const FwOpcodeType opCode, const U32 cmdSeq, Ref::ChoicePair choices) {
+    this->tlmWrite_ChoicePairCh(choices);
+    this->log_ACTIVITY_HI_ChoicePairEv(choices);
+    this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+}
+
+void TypeDemo ::CHOICE_PAIR_WITH_FRIENDS_cmdHandler(const FwOpcodeType opCode,
+                                                    const U32 cmdSeq,
+                                                    U8 repeat,
```
Comment (timcanham) [inline Ref/TypeDemo/TypeDemo.cpp:77]: repeat is never used?

File: Ref/TypeDemo/TypeDemo.cpp [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+void TypeDemo ::GLUTTON_OF_CHOICE_cmdHandler(const FwOpcodeType opCode, const U32 cmdSeq, Ref::ChoiceSlurry choices) {
+    this->tlmWrite_ChoiceSlurryCh(choices);
+    this->log_ACTIVITY_HI_ChoiceSlurryEv(choices);
+    this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+}
+
+void TypeDemo ::GLUTTON_OF_CHOICE_WITH_FRIENDS_cmdHandler(const FwOpcodeType opCode,
+                                                          const U32 cmdSeq,
+                                                          U8 repeat,
```
Comment (timcanham) [inline Ref/TypeDemo/TypeDemo.cpp:95]: repeat is never used?

**Review by timcanham — CHANGES_REQUESTED** [review]
Is it worth doing something with the unused arguments?

**Review by timcanham — APPROVED** [review]

**Pull request merged**